colcon_mixin/mixin/repository.py

```diff
@@ -16,8 +18,30 @@
 
 logger = colcon_logger.getChild(__name__)
 
-"""The path of the yaml file describing the mixin repositories."""
-mixin_repositories_file = get_config_path() / 'mixin_repositories.yaml'
+
+def get_mixin_repositories_file():
+    """
+    Get the path of the yaml file describing the mixin repositories.
+
+    :rtype: Path
+    """
+    return get_config_path() / 'mixin_repositories.yaml'
+
+
+if sys.version_info[:2] >= (3, 7):
+    def __getattr__(name):
+        if name == 'mixin_repositories_file':
+            warnings.warn(
+                "'colcon_mixin.mixin.repository.mixin_repositories_file' has "
+                "been deprecated, use 'colcon_mixin.mixin.repository."
+                "get_mixin_repositories_file()' instead", stacklevel=2)
+            return get_mixin_repositories_file()
+        raise AttributeError(
+            "module '%s' has no attribute '%s'" % (__name__, name))
```

Comment:
```suggestion
            f"module '{__name__}' has no attribute '{name}'")
```
Since the conditional checks for Python versions greater than 3.7, we can use f-strings instead.

Reply:
Even Python 3.6 has f-strings, this is just a copy/paste from older code. Regardless, :+1: